wallet/src/de/schildbach/wallet/ui/dashpay/DashPayViewModel.kt

```diff
@@ -85,13 +85,15 @@ open class DashPayViewModel(application: Application) : AndroidViewModel(applica
     // Search Usernames that start with "text".  Results are a list of documents for names
     // starting with text.  If no results are found then an empty list is returned.
     //
-    val searchUsernamesLiveData = Transformations.switchMap(userSearchLiveData) { search: Pair<String, Int> ->
+    val searchUsernamesLiveData = Transformations.switchMap(userSearchLiveData) { search: UserSearch ->
         searchUsernamesJob.cancel()
         searchUsernamesJob = Job()
         liveData(context = searchUsernamesJob + Dispatchers.IO) {
             emit(Resource.loading(null))
             try {
-                val result = platformRepo.searchUsernames(search.first, false, search.second)
+                var result = platformRepo.searchUsernames(search.text, false, search.limit)
+                result = result.filter {  !search.excludeIds.contains(it.dashPayProfile.userId) }
+                result = result.subList(0, 3)
```

Comment:
QA found this in the logs.  Perhaps 3 should be `min(result.size, 3)` - if there are only 1 or 2, then 3 will cause an exception
```
01-08 23:53:43.831 17099  4172 E DashPayViewModel: [DefaultDispatcher-worker-3] search usernames: toIndex = 3
01-08 23:53:43.832 17099  4172 E DashPayViewModel: [DefaultDispatcher-worker-3] toIndex = 3
01-08 23:53:43.832 17099  4172 W System.err: java.lang.IndexOutOfBoundsException: toIndex = 3
01-08 23:53:43.832 17099  4172 W System.err: 	at java.util.ArrayList.subListRangeCheck(ArrayList.java:1016)
```

Reply:
My above suggestion probably doesn't work when there are 0 results.

Reply:
Replacing line 96 with this seemed to work better.
```
                if (result.isNotEmpty()) {
                    val limit = result.size.coerceAtMost(search.limit)
                    result = result.subList(0, limit)
                }
```